isodatetime/data.py

```diff
@@ -1162,62 +1162,62 @@ def add_truncated(self, year_of_century=None, year_of_decade=None,
                       minute_of_hour=None, second_of_minute=None):
         """Combine this TimePoint with truncated time properties."""
         new = self.copy()
-        if hour_of_day is not None and minute_of_hour is None:
```

Comment:
I definitely agree there is a bug in the current implementation.

The main idea is that the timepoint is incremented until the data matches the truncated data... so the properties specified in the truncated timepoint should equal the ones in the final timepoint (assuming they are self-consistent).